presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -399,13 +400,16 @@ public synchronized void parentTasksAdded(List<TaskId> parentTasks, boolean noMo
                 newOutputBuffers = newOutputBuffers.withNoMoreBufferIds();
             }
         }
-        else if (fragment.getOutputPartitioning() == OutputPartitioning.HASH) {
+        else if (fragment.getOutputPartitioning() == OutputPartitioning.HASH || fragment.getOutputPartitioning() == OutputPartitioning.HASH_NULL_EVERYWHERE) {
             checkArgument(noMoreParentNodes, "Hash partitioned output requires all parent nodes be added in a single call");
 
+            NullPartitionPolicy nullPartitionPolicy = fragment.getOutputPartitioning() == OutputPartitioning.HASH ? NullPartitionPolicy.ONE_PLACE : NullPartitionPolicy.EVERYWHERE;
```

Comment:
This is brittle due to the fact that it can easily get out of sync with the `if` condition above and the possible values of the `OutputPartitioning` enum. Rewrite the assignment using a switch that throws if the default branch is taken:

```java
NullPartitioningPolicy nullPolicy;
switch (fragment.getOutputPartitioning()) {
    case HASH:
        nullPolicy = ONE_PLACE;
        break;
    case HASH_NULL_EVERYWHERE:
        nullPolicy = EVERYWHERE;
        break;
    default:
        throw new UnsupportedOperationException(...);
}
```